### PR TITLE
RavenDB-17082 Missing HasAttachments, HasCounters, HasTimeSeries flag…

### PR DESCRIPTION
**Pull request by efratshenhar**
…s in a reverted revision document

### CODE REVIEW

**Comment by ppekrol** [comment]
@efratshenhar Can you check failing tests?

**Comment by efratshenhar** [comment]
yes

File: src/Raven.Server/Documents/AttachmentsStorage.cs [hunk 1]
```diff
@@ -400,6 +400,16 @@ public string UpdateDocumentAfterAttachmentChange(DocumentsOperationContext cont
                 return UpdateDocumentAfterAttachmentChange(context, lowerDocumentId, documentId, tvr, null);
             }
         }
+        public void DeleteAttachmentDirect2(DocumentsOperationContext context, LazyStringValue lowerDocId)
```
Comment (karmeli87) [inline src/Raven.Server/Documents/AttachmentsStorage.cs]: please rename

File: src/Raven.Server/Documents/AttachmentsStorage.cs [hunk 2]
```diff
@@ -427,6 +437,37 @@ public void RevisionAttachments(DocumentsOperationContext context, Slice lowerId
...
+            {
+                hasAttachments = true;
+
+                if (attachment.TryGet(nameof(AttachmentName.Name), out LazyStringValue name) == false ||
+                    attachment.TryGet(nameof(AttachmentName.ContentType), out LazyStringValue contentType) == false ||
+                    attachment.TryGet(nameof(AttachmentName.Hash), out LazyStringValue hash) == false)
+                    throw new ArgumentException($"The attachment info in missing a mandatory value: {attachment}");
+
+                var cv = Slices.Empty;
+                var type = AttachmentType.Document;
```
Comment (karmeli87) [inline src/Raven.Server/Documents/AttachmentsStorage.cs:458]: why do we need those?

File: src/Raven.Server/Documents/Revisions/RevisionsStorage.cs [hunk 3]
```diff
@@ -1500,10 +1500,13 @@ protected override long ExecuteCmd(DocumentsOperationContext context)
                 foreach (var document in _list)
                 {
                     _token.ThrowIfCancellationRequested();
-
+                    CollectionName collectionName = RemoveOldMetadataInfo(context, documentsStorage, document);
```
Comment (karmeli87) [inline src/Raven.Server/Documents/Revisions/RevisionsStorage.cs]: do we need that if we revert to a tombstone?

File: src/Raven.Server/Documents/Revisions/RevisionsStorage.cs [hunk 4]
```diff
@@ -1519,6 +1522,30 @@ protected override long ExecuteCmd(DocumentsOperationContext context)
...
+                    metadata.TryGet(Constants.Documents.Metadata.RevisionCounters, out BlittableJsonReaderObject counters))
+                {
+                    var counterNames = counters.GetPropertyNames();
+
+                    foreach (var cn in counterNames)
+                    {
+                        var val = counters.TryGetMember(cn, out object value);
+                        documentsStorage.CountersStorage.PutCounter(context, document.Id, collectionName.Name, cn, (long)value);
+                    }
+                }
```
Comment (karmeli87) [inline src/Raven.Server/Documents/Revisions/RevisionsStorage.cs:1544]: Let's put it in a function and call it `RevertCounters`

**Review by karmeli87 — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Documents/AttachmentsStorage.cs (hunk 2 above)
Comment (efratshenhar) [reply src/Raven.Server/Documents/AttachmentsStorage.cs:458]: for GetAttachmentKey

File: src/Raven.Server/Documents/Revisions/RevisionsStorage.cs (hunk 3 above)
Comment (efratshenhar) [reply src/Raven.Server/Documents/Revisions/RevisionsStorage.cs]: no

File: test/SlowTests/Issues/RavenDB-17082.cs [hunk 5]
```diff
@@ -0,0 +1,613 @@
...
+                    await using (var backgroundStream = new MemoryStream(new byte[] { 10, 20, 30, 40, 50 }))
+                    {
+                        session.Advanced.Attachments.Store("foo/bar", "background-photo.jpg", backgroundStream, "ImGgE/jPeG");
+                        await session.SaveChangesAsync();
+                    }
+                }
+                WaitForUserToContinueTheTest(store);
+                using (var session = store.OpenAsyncSession())
+                {
+                    var persons = await session.Advanced.Revisions.GetForAsync<Person>("foo/bar");
```
Comment (karmeli87) [inline test/SlowTests/Issues/RavenDB-17082.cs:461]: In all of the tests you are verifying the revision... let's verify the document as well

**Review by karmeli87 — APPROVED** [review]
After merge, need to handle the time-series as well

**Pull request merged**